cmd/pulumi-test-language/tests/testdata/l1-resource-keyword-overlap/main.pp

```diff
@@ -0,0 +1,7 @@
+# Since the name is "this" it will fail in typescript and other languages with
+# this reservered keyword if it is not renamed.
+this = "somestring"
+
+output "output" {
+  value = this
+}
```

Comment:
_LOVE_ the idea of conformance testing this. I think we should get bang for our buck and add a load of "suspected reserved words" to this test. So e.g.:

```hcl
class = "class"
export = "export"
import = "import"
object = { object = "object" }
module = "module"
self = "self"
this = "this"

output "this_output" {
  value = this
}

...

output "all" {
  value = {
    ...
    this = this
  }
}
```

We could (should) also an an `l2-` test that does the same for some test `resource`s e.g. `resource "class" "..." { ... }`